### PR TITLE
[benchpress] Add Toplev hook

### PR DESCRIPTION
**Pull request by meteorfox**
Add support to running toplev either with the command (useful for short benchmarks) or in the background (useful for long running benchmarks).

Unit tests included

### CODE REVIEW

File: benchpress/benchpress/plugins/hooks/toplev.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+DEFAULT_TOPLEV_OUTPUT_CSV_PATH = 'toplev_output.csv'
+DEFAULT_OPTIONS = [
+    '-l1',  # Level
+    '-I100',
+    '--summary',
+    '-m',  # Extra metrics
+    '--no-desc',
+    '-x,',
+    '-o', DEFAULT_TOPLEV_OUTPUT_CSV_PATH,
+]
```
Comment (adhanotia) [inline benchpress/benchpress/plugins/hooks/toplev.py:29]: Do we assume that the event map files are downloaded on the system? Maybe add a comment in requirement section to download map files before launching the run with toplev hook.

File: benchpress/benchpress/plugins/hooks/toplev.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+import logging
+
+from benchpress.lib.hook import Hook
+from benchpress.lib import util
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_BACKGROUND_DURATION_SECS = '300'
```
Comment (adhanotia) [inline benchpress/benchpress/plugins/hooks/toplev.py:17]: 300s might be too long for some workloads like GAP which don't sustain steady state for that long. Maybe shorten the interval to 60s?

File: benchpress/benchpress/plugins/hooks/toplev.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        else:
+            binary_path = job.config['path']
+            job.args = opts['args'] + ['--', binary_path] + job.args
+            job.binary = opts.get('binary_path', DEFAULT_PATH)
+
+    def after_job(self, opts, job):
+        # Do nothing
+        pass
+
+    def _start_background_toplev(self, opts):
```
Comment (adhanotia) [inline benchpress/benchpress/plugins/hooks/toplev.py:64]: Do you plan to add the delayed start option separately? Or is that built in to the launch command?

**Review by adhanotia — CHANGES_REQUESTED** [review]
Looks great! Added some minor comments.

File: benchpress/benchpress/plugins/hooks/toplev.py (hunk 1 above)
Comment (meteorfox) [reply benchpress/benchpress/plugins/hooks/toplev.py:29]: Yes, we  are assuming the user already installed toplev and its dependencies. By default, as long as the host has a network connection to the internet, running toplev for the first time will automatically download the map files off from Intel's website.

File: benchpress/benchpress/plugins/hooks/toplev.py (hunk 2 above)
Comment (meteorfox) [reply benchpress/benchpress/plugins/hooks/toplev.py:17]: Sure sounds good.

File: benchpress/benchpress/plugins/hooks/toplev.py (hunk 3 above)
Comment (meteorfox) [reply benchpress/benchpress/plugins/hooks/toplev.py:64]: the delayed start is built-in into toplev using the `-D` flag

**Comment by adhanotia** [comment]
Looks good!

**Pull request merged**